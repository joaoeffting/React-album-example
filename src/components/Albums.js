//@flow

import React, { Component } from 'react';
import { Link } from 'react-router';
import {CommonService} from '../services/CommonService';
import './album.css';

type AlbumType = {
    id: number,
    title: string,
}

type State = {
    albums: Array<AlbumType>,
    loaded: boolean
}


class Albums extends Component {
    state: State = {
        albums: [],
        loaded: false
    }
    componentDidMount() {
        this.loadAlbums();
    }
    loadAlbums = async () => {
        try {
            let response = await CommonService.getAlbums();
            this.setState({albums: response.data, loaded: true});
        } catch(e) {
            console.log(e);
        }
    }
    render() {
        if (!this.state.loaded) return(<div>Carregando ... </div>);

        return (
            <div className="album-container">
                {
                    this.state.albums.map(album => {
                        let {id, title} = album;
                        return (
                            <Link to={`/photos/${id}`}>
                                <div key={id} className="album-item">
                                    {title}
                                </div>
                            </Link>
                        )
                    })
                }
            </div>
        );
    }
}

export default Albums;