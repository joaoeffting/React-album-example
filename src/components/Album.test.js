import renderer from 'react-test-renderer';
import React from 'react';
import Albums from './Albums';

test('Albums renders correctly', () => {
  const tree = renderer.create(
    <Albums />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});