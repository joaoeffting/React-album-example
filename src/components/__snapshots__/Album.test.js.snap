exports[`test Albums renders correctly 1`] = `
<div>
  Carregando ... 
</div>
`;
